### PR TITLE
API: Many item related endpoints added

### PR DESCRIPTION
**Pull request by annando**
Many API endpoints had been added for interaction with posts.

### CODE REVIEW

File: src/Module/Api/Mastodon/Statuses/Bookmark.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+			DI::mstdnError()->UnprocessableEntity();
+		}
+
+		$item = Post::selectFirstForUser($uid, ['id', 'gravity'], ['uri-id' => $parameters['id'], 'uid' => [$uid, 0]]);
+		if (!DBA::isResult($item)) {
+			DI::mstdnError()->RecordNotFound();
+		}
+
+		if ($item['gravity'] != GRAVITY_PARENT) {
+			DI::mstdnError()->UnprocessableEntity(DI::l10n()->t('Only starting post can be bookmarked'));
```
Comment (MrPetovan) [inline src/Module/Api/Mastodon/Statuses/Bookmark.php]:
```suggestion
			DI::mstdnError()->UnprocessableEntity(DI::l10n()->t('Only starting posts can be bookmarked'));
```

File: src/Module/Api/Mastodon/Statuses/Mute.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+			DI::mstdnError()->UnprocessableEntity();
+		}
+
+		$item = Post::selectFirstForUser($uid, ['id', 'gravity'], ['uri-id' => $parameters['id'], 'uid' => [$uid, 0]]);
+		if (!DBA::isResult($item)) {
+			DI::mstdnError()->RecordNotFound();
+		}
+
+		if ($item['gravity'] != GRAVITY_PARENT) {
+			DI::mstdnError()->UnprocessableEntity(DI::l10n()->t('Only starting post can be muted'));
```
Comment (MrPetovan) [inline src/Module/Api/Mastodon/Statuses/Mute.php]:
```suggestion
			DI::mstdnError()->UnprocessableEntity(DI::l10n()->t('Only starting posts can be muted'));
```

File: src/Module/Api/Mastodon/Statuses/Pin.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+			DI::mstdnError()->UnprocessableEntity();
+		}
+
+		$item = Post::selectFirstForUser($uid, ['id', 'gravity'], ['uri-id' => $parameters['id'], 'uid' => [$uid, 0]]);
+		if (!DBA::isResult($item)) {
+			DI::mstdnError()->RecordNotFound();
+		}
+
+		if ($item['gravity'] != GRAVITY_PARENT) {
+			DI::mstdnError()->UnprocessableEntity(DI::l10n()->t('Only starting post can be pinned'));
```
Comment (MrPetovan) [inline src/Module/Api/Mastodon/Statuses/Pin.php]:
```suggestion
			DI::mstdnError()->UnprocessableEntity(DI::l10n()->t('Only starting posts can be pinned'));
```

File: src/Module/Api/Mastodon/Statuses/Unbookmark.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+			DI::mstdnError()->UnprocessableEntity();
+		}
+
+		$item = Post::selectFirstForUser($uid, ['id', 'gravity'], ['uri-id' => $parameters['id'], 'uid' => [$uid, 0]]);
+		if (!DBA::isResult($item)) {
+			DI::mstdnError()->RecordNotFound();
+		}
+
+		if ($item['gravity'] != GRAVITY_PARENT) {
+			DI::mstdnError()->UnprocessableEntity(DI::l10n()->t('Only starting post can be unbookmarked'));
```
Comment (MrPetovan) [inline src/Module/Api/Mastodon/Statuses/Unbookmark.php]:
```suggestion
			DI::mstdnError()->UnprocessableEntity(DI::l10n()->t('Only starting posts can be unbookmarked'));
```

File: src/Module/Api/Mastodon/Statuses/Unmute.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+			DI::mstdnError()->UnprocessableEntity();
+		}
+
+		$item = Post::selectFirstForUser($uid, ['id', 'gravity'], ['uri-id' => $parameters['id'], 'uid' => [$uid, 0]]);
+		if (!DBA::isResult($item)) {
+			DI::mstdnError()->RecordNotFound();
+		}
+
+		if ($item['gravity'] != GRAVITY_PARENT) {
+			DI::mstdnError()->UnprocessableEntity(DI::l10n()->t('Only starting post can be unmuted'));
```
Comment (MrPetovan) [inline src/Module/Api/Mastodon/Statuses/Unmute.php]:
```suggestion
			DI::mstdnError()->UnprocessableEntity(DI::l10n()->t('Only starting posts can be unmuted'));
```

File: src/Module/Api/Mastodon/Statuses/Unpin.php [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+			DI::mstdnError()->UnprocessableEntity();
+		}
+
+		$item = Post::selectFirstForUser($uid, ['id', 'gravity'], ['uri-id' => $parameters['id'], 'uid' => [$uid, 0]]);
+		if (!DBA::isResult($item)) {
+			DI::mstdnError()->RecordNotFound();
+		}
+
+		if ($item['gravity'] != GRAVITY_PARENT) {
+			DI::mstdnError()->UnprocessableEntity(DI::l10n()->t('Only starting post can be pinned'));
```
Comment (MrPetovan) [inline src/Module/Api/Mastodon/Statuses/Unpin.php]:
```suggestion
			DI::mstdnError()->UnprocessableEntity(DI::l10n()->t('Only starting posts can be pinned'));
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Review by MrPetovan — APPROVED** [review]

**Comment by AlfredSK** [comment]
Composer deprecation warning after pulling this one.

```
Deprecation Notice: Class Friendica\Module\Api\Mastodon\Accounts\Relationships located in ./src/Module/Api/Mastodon/Accounts/Relationsships.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///.../.../bin/composer.phar/src/Composer/Autoload/ClassMapGenerator.php:201
```

**Comment by annando** [comment]
@AlfredSK will disappear after PR https://github.com/friendica/friendica/pull/10250 had been merged.

**Pull request merged**